XenAdmin/Commands/MoveVMCommand.cs

```diff
@@ -67,26 +67,23 @@ protected override void ExecuteCore(SelectedItemCollection selection)
         {
             VM vm = (VM)selection[0].XenObject;
 
-            MainWindowCommandInterface.ShowPerXenModelObjectWizard(vm, new MoveVMDialog(vm));
+            var cmd = new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection);
+            if (cmd.CanExecute())
+                cmd.Execute();
+            else
+                MainWindowCommandInterface.ShowPerXenModelObjectWizard(vm, new MoveVMDialog(vm));
         }
 
         protected override bool CanExecuteCore(SelectedItemCollection selection)
         {
+            if (new CrossPoolMoveVMCommand(MainWindowCommandInterface, selection).CanExecute())

```

Comment:
I am afraid this (creating new objects) will run too many times. 
This could be solved by a static method if it fits here